### ISSUE TITLE
No sources shown in dropdown list

### ISSUE THREAD

**Issue by nitsnatsnok** [lingdb/CoBL#354]
Since release [2016-11-22.0](https://github.com/lingdb/CoBL/releases/tag/2016-11-22.0) the dropdown list doesn't show a single source when entering a new citation (to an existing source). All citations are affected by this, so effectively it's not possible to add a citation at the moment.

![2016-11-22-135151_1366x768_scrot](https://cloud.githubusercontent.com/assets/2664162/20524574/5573162c-b0bb-11e6-8e2a-34e6fb125833.png)

**Comment by khoidt** [comment]
I'm on it.

**Comment by khoidt** [comment]
@runjak it would be probably better to get back to the previous release for now.

**Comment by nitsnatsnok** [comment]
Unless there's a really quick fix that seems the only way – we're in the middle of the workshop!

**Comment by khoidt** [comment]
It seems to work locally:
![autocomplete](https://cloud.githubusercontent.com/assets/7487683/20525457/e34ba06e-b0bf-11e6-805e-5f24aff1c0bc.jpg)

**Comment by khoidt** [comment]
The server reports the following javascript errors:
```
Uncaught Error: See almond README: incorrect module build, no module name
    at define (http://cobl.info/ielex/static/minified.96328743.js:34:2778)
    at http://cobl.info/ielex/static/autocomplete_light/vendor/select2/dist/js/select2.full.js:11:5
    at http://cobl.info/ielex/static/autocomplete_light/vendor/select2/dist/js/select2.full.js:19:2
select2.js:47 Uncaught TypeError: $(...).select2 is not a function(…)
```

**Comment by khoidt** [comment]
I believe this has to do with setting `select2.js` with AMD (probably, [this](https://select2.github.io/options.html#select2-is-being-placed-before-jquery-in-my-javascript-file), see also [this](https://github.com/yourlabs/django-autocomplete-light/issues/641)). I'll try to handle it, but as I'm new to AMD, it might take me a while. @runjak, let me know, please if you understand how to solve this.

**Comment by runjak** [comment]
I think the best fix for the AMD situation would be to include the `select2` dependency in the AMD dependencies so that it will also be delivered in the minified .js file. It would make sense to also manage this dependency with bower. I've already got my hands on the corresponding bower module, but adding the dependency to AMD and informing the django-autocomplete-light module of this may need a bit of work still.

**Comment by runjak** [comment]
This should be fixed now. @khoidt you can do with this issue what you want .)

**Comment by khoidt** [comment]
Thank you! It works on the server, so I'm closing the issue.

**Comment by nitsnatsnok** [comment]
I confirm that this works like a charm now, excellent work you both! :tada:
There is a small issue that the comment field is too wide on the right:
![2016-11-23-212037_1366x768_scrot](https://cloud.githubusercontent.com/assets/2664162/20577509/f7a89cc0-b1c2-11e6-94f2-6f039bafeb76.png)

But this is neither curcial nor does it belong here.

**Comment by khoidt** [comment]
@nitsnatsnok, add this to #344, please!

**Comment by nitsnatsnok** [comment]
As #344 is already quite confusing, I'm just opening up a new low-priority issue for this (#357) and add a link in #344.

**Comment by nitsnatsnok** [comment]
Unfortunately, we were too happy too soon: The dropdown list works for lexeme as well as cognacy citations, but not so for cognate class citations:
![2016-11-24-110634_1366x768_scrot](https://cloud.githubusercontent.com/assets/2664162/20594352/2fb937c8-b236-11e6-94de-68834fb623a3.png)

**Comment by khoidt** [comment]
## Update
- Change `generic_update.html` template to fix cognate class citation issue in #354.
---
@runjak, in #306. Deploy it, please!

**Comment by runjak** [comment]
@khoidt why are you adding the `{{ form.media }}` directive back in 5f123c0c302fe28e43d2571ff8237ba216073b48? I've just removed it in the latest commits in `devel` which worked fine because the corresponding JavaScript is now included trough our own asset pipeline instead of trough the `django-autocomplete-light` module.

**Comment by runjak** [comment]
Ah, sorry - I see this is about the `generic_update.html` instead of the `lexeme_report.html` one. But still I've just removed this same code from a different template before.

**Comment by khoidt** [comment]
@runjak, sorry, I didn't notice that you removed it.

**Comment by runjak** [comment]
Yes - the code it includes handles inclusion of jQuery in a way that breaks.
The other way around we can roll everything into a single minified `.js` file that the client can cache. Because the minified file is now named in part by its hash this doesn't bring caching problems.

The more interesting thing would be: Does the thing you've been trying to achieve with the commit actually work in the current site or am I missing something here?

**Comment by khoidt** [comment]
Yes, locally it does work (at least, when `django-debug-toolbar` is disabled):
![capture](https://cloud.githubusercontent.com/assets/7487683/20599912/1b88a282-b251-11e6-84f5-bba6b18354ce.JPG)

**Comment by runjak** [comment]
Hm. Strange - I would have expected for this to already work on the server as well.
Investigating…

**Comment by runjak** [comment]
Hm. it's working in 2bec9c09ddd9214ccf798302c36fca989719d4fc (`devel`) on my machine for a route like `/cognate/178/add-citation/` but I see that it doesn't work remote at the moment.

**Comment by khoidt** [comment]
I revised the updated templates, and if I understand the piping logic right, the template needs  `{% load staticfiles %}` or/and ` {% load static %}` to work on the server. `lexeme_report.html` had neither.

**Comment by khoidt** [comment]
`generic_update.html` of course, not `lexeme_report.html`.

**Comment by runjak** [comment]
`base.html` should already include a file named `minified.d14bca71.js` or similar which should handle this. From looking at the browser debug tools I see that this breaks on the server while working fine locally.

**Comment by runjak** [comment]
That is I get a complaint but later on `$.fn.select2` is actually defined:
![screenshot_2016-11-24_14-40-22](https://cloud.githubusercontent.com/assets/5417642/20600445/fd6d2ab8-b253-11e6-8bac-674075e5fb25.png)
I think I'm mistakenly including some scripts in the wrong order or something like that so that one module doesn't find the function it's looking for but later that module is actually loaded.

**Comment by khoidt** [comment]
If so, it might have to do with
> Ensure that jquery is loaded before `{{ form.media }}`

From [django-autocomltere-light3 tutorial](http://django-autocomplete-light.readthedocs.io/en/master/tutorial.html#using-autocompletes-outside-the-admin).

**Comment by runjak** [comment]
Sorry, I've been a bit slow on this one figuring out the problem.
What I found was this:
The `minified.js` file wasn't included correctly in the template because of two things:
1. When `Debug=True` the non minified wasn't used as a fallback.
2. Because the `generic_update.html` template is used by one of the class views and their rendering uses a specialized context the `base.html` didn't end up using the `minified.js` file even though it was present.

Both of these have been addressed in e9c44801e4506ca2faf8b563505f1b9ace910167, which will be deployed soon.
